provider/openstack/provider.go

```diff
@@ -1155,15 +1155,25 @@ func (e *Environ) StartInstance(ctx context.ProviderCallContext, args environs.S
 			} else if serverDetail.Status == nova.StatusActive {
 				break
 			} else if serverDetail.Status == nova.StatusError {
+				if server == nil {
+					break
```

Comment:
We're in deeper trouble if this is true.  Suggest moving to before 1150 and set err in this case, otherwise we'll try to use 'server' at ln 1208.